src/main/java/com/selfxdsd/storage/SelfPms.java

```diff
@@ -74,34 +72,74 @@ public ProjectManager getById(final int projectManagerId) {
                 .where(SLF_PMS_XDSD.ID.eq(projectManagerId))
                 .fetch();
             if(result.size() > 0) {
-                final Record rec = result.get(0);
-                final ProjectManager found = new StoredProjectManager(
-                    rec.getValue(SLF_PMS_XDSD.ID),
-                    rec.getValue(SLF_PMS_XDSD.PROVIDER),
-                    rec.get(SLF_PMS_XDSD.ACCESS_TOKEN),
-                    this.storage
-                );
-                return found;
+                return mapToProjectManager(result.get(0));
             }
         }
         return null;
     }
 
     @Override
     public ProjectManager pick(final String provider) {
-        throw new UnsupportedOperationException("Not yet implemented.");
+        try (final Database connected = this.database.connect()) {
+            final Result<Record> result = connected.jooq()
+                .select()
+                .from(SLF_PMS_XDSD)
+                .where(SLF_PMS_XDSD.PROVIDER.eq(provider))
+                .limit(1)
+                .fetch();
+            if(result.size() > 0) {
+                return this.mapToProjectManager(result.get(0));
+            }
+        }
+        return null;
     }
 
     @Override
     public ProjectManager register(
         final String provider,
-        final String accessToken
-    ) {
-        throw new UnsupportedOperationException("Not yet implemented.");
+        final String accessToken) {
+        try (final Database connected = this.database.connect()) {
+            final int pmId = connected.jooq()
+                .insertInto(SLF_PMS_XDSD,
+                    SLF_PMS_XDSD.PROVIDER,
+                    SLF_PMS_XDSD.ACCESS_TOKEN)
+                .values(provider, accessToken)
+                .execute();
+            if(pmId > 0){
+                return new StoredProjectManager(pmId,
+                    provider,
+                    accessToken,
+                    storage);
+            }
+        }
+        throw new IllegalStateException("Something went wrong while"
+            + " inserting into database.");
     }
 
     @Override
     public Iterator<ProjectManager> iterator() {
-        throw new UnsupportedOperationException("Not yet implemented.");
+        try (final Database connected = this.database.connect()) {
+            return connected.jooq()
+                .select()
+                .from(SLF_PMS_XDSD)
+                .fetch()
```

Comment:
@criske SelfPMs represents *all* the PMs in Self. There will not be a lot of PMs, maybe 2 or 3 per Provider but even so, can you please specify ``limit(100)`` here?